src/main/java/xyz/thepathfinder/android/Cluster.java

```diff
@@ -244,31 +270,52 @@ public Cluster getSubcluster(String name) {
 
     /**
      * Returns an immutable collection of this cluster's direct subclusters.
+     *
      * @return An immutable collection of clusters.
      */
     public Collection<Cluster> getSubclusters() {
```

Comment:
I use Collection because internally it is HashMap of the model, You can make it an immutable set very easily, but a list will not do much. All a user should do is iterate over it, so there is very little gained from any other data structure.